spec/javascript/components/axis_spec.js

```diff
@@ -0,0 +1,63 @@
+import { Axis } from '../../../app/javascript/components/axis'
+
+describe('components/axis.js', function() {
+  function createAxis(properties) {
+    if (properties === undefined) {
+      properties = {}
+    }
+    var defaults = { name: "q", degrees: 60, xOffset: 13, yOffset: 94, hexDiameter: 133, mapWidth: 800, mapHeight: 800, dashFilledLength: 44, dashBlankLength: 89, color: "#ff0000" }
```

Comment:
It doesn't feel quite right for an Axis to have a `mapWidth` and `mapHeight`. Perhaps a `canvas` `area` or `section` or something more general?

Reply:
This will probably be handled in a future refactor.

Reply:
I think this is fine for now